internal/authorization/authorization.go

```diff
@@ -0,0 +1,40 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+)
+
+const (
+	DriverNoop    string = "noop"
+	DriverOpenFGA string = "openfga"
+)
+
+var authorizers = map[string]func() Authorizer{
+	DriverNoop:    func() Authorizer { return &NoopAuthorizer{} },
+	DriverOpenFGA: func() Authorizer { return &OpenFGAAuthorizer{} },
+}
```

Comment:
have a look at https://go.dev/play/p/5V3KqjuVKTp

basically using a custom type definition and then define consts with well defined names